app/src/lib/promise.ts

```diff
@@ -37,3 +37,18 @@ export function promiseWithMinimumTimeout<T>(
       .catch(reject)
   })
 }
+
+/**
+ * `async`/`await`-friendly wrapper around `window.setTimeout` for places where
+ * callers want to defer async work and avoid the ceremony of this setup and
+ * using callbacks
+ *
+ * @param timeout the time to wait before resolving the promise (in milliseconds)
+ */
+export async function timeout(timeout: number): Promise<void> {
```

Comment:
This helper function was a neater way to support deferring asynchronous work. I have some upcoming changes when I switch between dialogs that made this sort of code a bit more readable:

```ts
  public async componentDidMount() {
    if (this.props.onDidMount) {
      await timeout(500)
      await this.props.onDidMount()
    }
  }
```

Reply:
This is neat, I'll just note I would rather see us rely less on `setTimeout` (except for the cases where css animation is involved since i don't know of a a better solution). This isn't blocking though, since I know we have a number of these in the codebase already.

Out of curiousity, @shiftkey is the above example a place you're planning to use this?

Reply:
Yes. I'm open to other alternatives to achieve this, but ultimately I need some way to perform asynchronous work _after_ the component is mounted and displayed for the first time.